### ISSUE TITLE
Editing of allowed / deny tags does not work

### ISSUE THREAD

**Issue by sker65** [janeczku/calibre-web#1788]
**Describe the bug/problem**
When I try to edit per user tags, the UI always only shows "my tags" (the admin ones) but not the tags from the user that is edited. when I double check in the app.db the tags is indeed in the admin record

**To Reproduce**
1. Go to 'Admin'
2. Click on 'one of the users other than admin'
3. Scroll down to 'Add allow / deny tag'
4. You will see the exact same deny / allow tag under admin user
5. It filters in admin users view but not in the view of the intended user

**Expected behavior**
Should be editable per user from the admin and also filter by that users tags, not admin tags

**Screenshots**

**Environment (please complete the following information):**
 
 - Calibre-Web version: 0.6.10 beta
 - Docker container: [/Technosoft2000/]:

**Comment by OzzieIsaacs** [comment]
I'm not using docker container, but with my normal installation (newest master branch 10days ago) it works:
I started with a new installation. Went to admin edit user (admin user). Scrolled down, clicked on `Add Allowed/Denied tags`
![grafik](https://user-images.githubusercontent.com/6438895/103403068-3b0adf80-4b4f-11eb-83f2-b4150806f372.png)
Added a random string and clicked on deny:
![grafik](https://user-images.githubusercontent.com/6438895/103403090-4958fb80-4b4f-11eb-87b7-7a0586d9bc28.png)
Created a new user, saved the new user, edited the new user, checked the add deny columns, they where emtpy, added a new string:
![grafik](https://user-images.githubusercontent.com/6438895/103403170-8e7d2d80-4b4f-11eb-86e2-cda8d2cd7c45.png)
Checked it wih the admin account, nothing changed:
![grafik](https://user-images.githubusercontent.com/6438895/103403250-c5534380-4b4f-11eb-8a15-5ce0a5c0037b.png)
Please be sure that you did not add accidently some tags in the "user template" section, which might confuse you:
![grafik](https://user-images.githubusercontent.com/6438895/103403294-e9168980-4b4f-11eb-997e-0f5a5839d059.png)

I could offer you one more thing: In the admin section there is a new button, "Download Debug Package", this includes all log files and a copy of the settings database (without passwords), you could send me this one and I could double check with exactly the same settings you have. You could send me the file via private email (email address given on my profile page on github)

**Comment by sker65** [comment]
I tracked the error further down, looking at the code of admin.py e.g. add_restriction handler, it seems that it depends on the referer header in the request to use "the correct" user. In my case this does not work as I use a docker container with nginx as reverse proxy as frontend that also handles TLS stuff.
The Referer policy is set to strict-origin, but if the flask service behind the nginx only see's "local" requests it does not work.
I fixed it by adding some header rewrite magic in nginx config, but in general I don't think its good practice to rely on referer headers being set instead of passing request params directly in the uri, query or request body.
What I added to nginx:
                proxy_hide_header       Referer-Policy;
                add_header              Referer-Policy unsafe-url;
This way referer gets set in the expected way and it works, even so this referer policy is kind of unsecure, so I would recommened to stay away from using referer header as request param for handler business logic in general. In my mind this is not what the referer header is meant for.
Br Stefan
